packages/relay/src/lib/eth.ts

```diff
@@ -669,12 +669,38 @@ export class EthImpl implements Eth {
                   currentTimestamp = mirrorAccount.balance.timestamp;
                 }
 
-                let transactionsInTimeWindow = await this.mirrorNodeClient.getTransactionsForAccount(
-                  mirrorAccount.account,
-                  block.timestamp.to,
-                  currentTimestamp,
-                  requestId
-                );
+                // Need to check if there are any transactions before the block.timestamp.to in the current account set returned from the inital
+                // call to getAccountPageLimit.  If there are we may need to paginate.
+                let lastTransactionOnPageTimestamp;
+                if(mirrorAccount.links.next !== null) {
+                  // Get the end of the page of transactions timestamp
+                  const params = new URLSearchParams(mirrorAccount.links.next.split('?')[1]);
+                  if((params === null) || (params === undefined)) {
+                    this.logger.debug(`${requestIdPrefix} Unable to find expected search parameters in account next page link ${mirrorAccount.links.next}), returning 0x0 balance`);
+                    return EthImpl.zeroHex;
+                  }
+
+                  const timestampParameters = params.getAll('timestamp'); 
+                  lastTransactionOnPageTimestamp = timestampParameters[0].split(':')[1];
+                  if((lastTransactionOnPageTimestamp === null) || (lastTransactionOnPageTimestamp === undefined)) {
+                    this.logger.debug(`${requestIdPrefix} Unable to find expected beginning (gte:) timestamp in account next page link ${mirrorAccount.links.next}), returning 0x0 balance`);
+                    return EthImpl.zeroHex;
+                  }
+                }
+
+                let transactionsInTimeWindow: any = [];
+                if((typeof lastTransactionOnPageTimestamp !== "undefined") && (mirrorAccount.transactions[mirrorAccount.transactions.length -1].consensus_timestamp >= lastTransactionOnPageTimestamp)) {
```

Comment:
SO this is an example of the last transaction and a valid next link
```
{"bytes":null,"charged_tx_fee":169304,"consensus_timestamp":"1682378101.517897720","entity_id":null,"max_fee":"100000000","memo_base64":"RGV2T3BzIFN5bnRoZXRpYyAyLjAgVGVzdGluZw==","name":"CRYPTOTRANSFER","node":"0.0.8","nonce":0,"parent_consensus_timestamp":null,"result":"SUCCESS","scheduled":false,"staking_reward_transfers":[],"transaction_hash":"1iSh8lSUcEXcdsydXm+9CCq5cUodnTKUR4JfCb2BMU5YF6kaU3QintjF/m91l2iP","transaction_id":"0.0.90-1682378091-167906093","transfers":[{"account":"0.0.2","amount":1,"is_approval":false},{"account":"0.0.8","amount":6866,"is_approval":false},{"account":"0.0.90","amount":-169305,"is_approval":false},{"account":"0.0.98","amount":162438,"is_approval":false}],"valid_duration_seconds":"120","valid_start_timestamp":"1682378091.167906093"}],"links":{"next":"/api/v1/accounts/0.0.2?timestamp=lt:1682378101.517897720"}}
```

By definition the last transaction timestamp and the next timestamp value should be equal, so maybe you only have to check for a valid next.

However in this case it's still not clear to me what's being checked for.

What does the current time stamp refer to in this flow and how come this query doesn't use the transaction next timestamp.
Is there a timestamp gap here?